packages/@angular/cli/tasks/serve.ts

```diff
@@ -31,7 +31,15 @@ export default Task.extend({
     if (projectConfig.project && projectConfig.project.ejected) {
       throw new SilentError('An ejected project cannot use the build command anymore.');
     }
-    rimraf.sync(path.resolve(this.project.root, outputPath));
+
+    const contentBase = path.resolve(this.project.root, outputPath);
+    if (serveTaskOptions.useDist) {
+      if(!fs.existsSync(contentBase)) {
+        throw new SilentError('Dist directory not found');
```

Comment:
If the point of this is to serve whatever got built in `build`, I think this would be explicitly outside of the purpose of this command.